### ISSUE TITLE
Broken Merge Tag Insertion

### ISSUE THREAD

**Issue by wpnzach** [wpninjas/ninja-forms#2740]
When merge tags with Capital Letters are inserted, subsequent mergetags override the previous merge tag.

Steps to reproduce (with Stripe):
1) Add `{stripe:chargeID}`
1) Add `{stripe:customerID}`
1) Expect `{stripe:chargeID}{stripe:customerID}`
1) See `{stripe:customerID}ID}`

Edit: Re-purposing this issue for broken merge tag insertion. Cursor placement issues in RTE moved to #2883

----
~~When inserting merge tags into an RTE with multiple lines, the cursor position jumps around in the RTE, which eventually breaks inserting new merge tags.~~

~~https://db.tt/x0yeOEpNXm~~

~~Not sure if this is exclusive to calk merge tags or not.  The video above does a good job detailing the issues.  Related to https://wpninjas.kustomerapp.com/app/customers/591496a2e041380010cbffaf/event/591496a2e041380010cbffb4~~

**Comment by wpnzach** [comment]
Please see this video, starting at one minute in.  This explains the majority of the usability issues customers are currently facing with our merge selector: https://vimeo.com/225419908

**Comment by kjohnson** [comment]
@wpnzach are you able to reproduce this locally? I'm not able to replicate the behavior in the video.

**Comment by wpnzach** [comment]
Yes, consistently.  @jameslaws has also seen these behaviors.  I just tested again before commenting.  Would you like a video from my machine as well?

**Comment by kjohnson** [comment]
I can see it happening in the video, but I have not been unable to reproduce it.

**Comment by wpnzach** [comment]
Which part are you unable to reproduce?  There are a few behaviors in the video that need addressed?

**Comment by kjohnson** [comment]
Currently, I'm not seeing any issue with merge tag insertion with the RTE.

Trying to reproduce it. Next I'm trying specifically with stripe, in case that is the issue.

**Comment by kjohnson** [comment]
Hm, ok. Seeing it with Stripe merge tags, specifically.

**Comment by kjohnson** [comment]
Looks like it may be specific to the `chargeID` and `customerID` merge tags.
Only difference that I see there would be Capitalization.
Maybe there is a regex issue?

Could also explain the issue with the Query String merge tag, which uses capital letters.

**Comment by kjohnson** [comment]
Field Keys do not allow for Capitalization, which means that if Capitalization is the issue, then it would not be seen using field merge tags.

**Comment by wpnzach** [comment]
Capitalization can happen in calc names as well and in common, so that could be it.

Positioning of the merge tag selector is another big issue as part of this.  Another issue, which i can seperate if we want, is inability to scroll in the RTE and being pushed to the top whenever i add a merge tag that would be outside (below) the current RTE area.  So if it's a long merge tag in the last visible line, when it adds im autoscrolled to the top of the RTE and I can't scroll back down without vertically expanding the RTE

**Comment by kjohnson** [comment]
Yeah, there are some complications with the RTE in that it is a Content Editable `div`, as opposed to a textarea. The cursor position has no context of the parent element and adding a new line is actually creating a new child element.

I'll look for a better way to handle that.

That said, I was able to correct the broken merge tag insertion, which was caused by a case sensitive regex pattern. This is why I wasn't able to reproduce the issue with fields; field keys do not allow capital letters. The reproduction specifically required merge tags with capital letters, ie Stripe.
